scio-avro/src/main/scala/com/spotify/scio/avro/taps.scala

```diff
@@ -26,22 +26,36 @@ import com.spotify.scio.values._
 import com.twitter.chill.Externalizer
 import org.apache.avro.Schema
 import org.apache.avro.generic.GenericRecord
+import org.apache.avro.specific.SpecificRecordBase
 
 import scala.concurrent.Future
 import scala.reflect.ClassTag
 import scala.reflect.runtime.universe._
 
 /**
- * Tap for Avro files.
- * @param schema must be not null if `T` is of type
- *               [[org.apache.avro.generic.GenericRecord GenericRecord]].
+ * Tap for [[org.apache.avro.generic.GenericRecord GenericRecord]] Avro files.
  */
-case class AvroTap[T: ClassTag: Coder](path: String, @transient private val schema: Schema = null)
+case class SchemaAvroTap[T: ClassTag: Coder](path: String,
```

Comment:
i kind of feel like this should be named `GenericRecordTap` to match `SpecificRecordTap` ? what do you think?

Reply:
Yeah, makes sense.